src/commonMain/kotlin/org/kbson/serialization/Serializers.kt

```diff
@@ -0,0 +1,799 @@
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonArray
+import kotlinx.serialization.json.JsonDecoder
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonEncoder
+import kotlinx.serialization.json.JsonNull
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.booleanOrNull
+import kotlinx.serialization.json.contentOrNull
+import kotlinx.serialization.json.doubleOrNull
+import kotlinx.serialization.json.floatOrNull
+import kotlinx.serialization.json.intOrNull
+import kotlinx.serialization.json.longOrNull
+import kotlinx.serialization.modules.SerializersModule
+import org.kbson.BsonArray
+import org.kbson.BsonBinary
+import org.kbson.BsonBoolean
+import org.kbson.BsonDBPointer
+import org.kbson.BsonDateTime
+import org.kbson.BsonDecimal128
+import org.kbson.BsonDocument
+import org.kbson.BsonDouble
+import org.kbson.BsonInt32
+import org.kbson.BsonInt64
+import org.kbson.BsonJavaScript
+import org.kbson.BsonJavaScriptWithScope
+import org.kbson.BsonMaxKey
+import org.kbson.BsonMinKey
+import org.kbson.BsonNull
+import org.kbson.BsonObjectId
+import org.kbson.BsonRegularExpression
+import org.kbson.BsonSerializationException
+import org.kbson.BsonString
+import org.kbson.BsonSymbol
+import org.kbson.BsonTimestamp
+import org.kbson.BsonType
+import org.kbson.BsonUndefined
+import org.kbson.BsonValue
+import org.kbson.internal.Base64Utils
+import org.kbson.internal.HexUtils
+import org.kbson.internal.validateSerialization
+
+/**
+ * The BsonSerializersModule
+ *
+ * Contains serializers that handle the conversion of Bson data to extended json and back
+ */
+public val BsonSerializersModule: SerializersModule = SerializersModule {
+    contextual(BsonValue::class, BsonValueSerializer)
+}
+
+
+/**
+ * The Bson companion object
+ */
+public object Bson {
+
+    private val extendedJson = Json { BsonSerializersModule }
+
+    /**
+     * Create a BsonDocument from a Json string
+     *
+     * @param json the Json String
+     * @return a BsonDocument
+     */
+    public operator fun invoke(jsonString: String): BsonValue {
+        return extendedJson.decodeFromString(BsonValueSerializer, jsonString)
+    }
+
+    /**
+     * Create a Json string from a BsonValue
+     *
+     * @param bsonValue the BsonValue
+     * @return the Json String
+     */
+    public fun toJson(bsonValue: BsonValue): String {
+        return extendedJson.encodeToString(BsonValueSerializer, bsonValue)
+    }
+}
+
+@OptIn(ExperimentalSerializationApi::class)
```

Comment:
I'm not sure how big the risk is of opting in to experimental API.  It it worth it?

Reply:
All serializer API's are experimental right now: https://kotlinlang.org/docs/serialization.html#formats, so there is really no way around it.

It also just means that the API's might change/not be backwards compatible rather than they have bugs. Since Multiplatform support in general is alpha I think this is okay.